### PR TITLE
Add order number to checkout schema

### PR DESCRIPTION
**Pull request by hsingyuc**
The issue is the sequential order number plugin is not compatible with WooPay.  Because we don't pass the order number in the checkout endpoint so WooPay can only use the order id as the order number.

### Testing
1. Preserve log and go through the Block checkout
2. Find the checkout response and see the order number
3. Smoke test the blocks checkout to make sure nothing is broken

### Changelog

> Add order number to checkout schema.

### CODE REVIEW

**Review by malithsen — APPROVED** [review]
@hsingyuc LGTM. I see the `order_number` field in the checkout response and didn't notice any issues when smoke testing. Let's also get a review from Rubik

**Comment by hsingyuc** [comment]
@ralucaStan  Hi Raluca! I'm not sure how you assign reviewers on Rubik team, so I pin you here and let you assign the reviewer for this PR.  Thank you for your help!

**Comment by ralucaStan** [comment]
Hey @hsingyuc, 
you just need to select `woocommerce/Rubik` and that will assign a team member for review. I already did this for this PR
Screenshot 2023-06-22 at 10 47 53

**Review by mikejolley — APPROVED** [review]
👍🏻 LGTM Re-ran test suite—let's make sure its green before merge.

**Pull request merged**
